crates/sargon/src/signing/collector/signatures_collector.rs

```diff
@@ -0,0 +1,2294 @@
+                        &factor_source.factor_source_id().as_hash().unwrap(), // TODO check that
+                    );
+
+                    if !request.invalid_transactions_if_neglected.is_empty() {
+                        info!(
+                            "If factor {:?} are neglected, invalid TXs: {:?}",
+                            request.input.factor_source_id,
+                            request.invalid_transactions_if_neglected
+                        )
+                    }
+
+                    debug!(
+                        "Dispatching mono request to interactor: {:?}",
+                        request
+                    );
+                    // Produce the results from the interactor
+                    let response = interactor.sign(request).await;
+                    debug!("Got response from mono interactor: {:?}", response);
+
+                    // Report the results back to the collector
+                    self.process_batch_response(response);
+
+                    if self.continuation() == FinishEarly {
+                        break;
+                    }
+                }
+            }
+        }
+    }
+
+    /// In decreasing "friction order"
+    async fn sign_with_factors(&self) -> Result<()> {
+        let factors_of_kind = self.dependencies.factors_of_kind.clone();
+        for factor_sources_of_kind in factors_of_kind.iter() {
+            if self.continuation() == FinishEarly {
+                break;
+            }
+            if self.neglected_factors_due_to_irrelevant(factor_sources_of_kind)
+            {
+                continue;
+            }
+            self.sign_with_factors_of_kind(factor_sources_of_kind).await;
+        }
+        info!("FINISHED WITH ALL FACTORS");
+        Ok(())
+    }
+
+    fn input_for_interactor(
+        &self,
+        factor_source_id: &FactorSourceIDFromHash,
+    ) -> MonoFactorSignRequestInput {
+        self.state
+            .borrow()
+            .petitions
+            .borrow()
+            .input_for_interactor(factor_source_id)
+    }
+
+    fn request_for_serial_interactor(
+        &self,
+        factor_source_id: &FactorSourceIDFromHash,
+    ) -> MonoFactorSignRequest {
+        let batch_signing_request = self.input_for_interactor(factor_source_id);
+
+        MonoFactorSignRequest::new(
+            batch_signing_request,
+            self.invalid_transactions_if_neglected_factor_sources(
+                IndexSet::just(*factor_source_id),
+            )
+            .into_iter()
+            .collect::<IndexSet<_>>(),
+        )
+    }
+
+    fn request_for_parallel_interactor(
+        &self,
+        factor_sources_of_kind: &FactorSourcesOfKind,
+    ) -> PolyFactorSignRequest {
+        let factor_source_ids = factor_sources_of_kind
+            .factor_sources()
+            .iter()
+            .map(|f| f.factor_source_id().as_hash().unwrap().clone()) // TODO check that
```

Comment:
TODO:
The same should apply here

Reply:
see https://github.com/radixdlt/sargon/pull/235#discussion_r1796762949